### PR TITLE
Order checkout stock lock

### PR DESCRIPTION
**Pull request by RUKAIYIN**
Add read/write when checking the availability of stock items and the test as well. But there's no test for insufficient stocks for concurrent orders. The results of this situation are unstable and I don't have a good idea of how to test it. I'd like to add a test for it or improve the read/write lock in the next phase.

### CODE REVIEW

File: service/stock/src/test/java/nl/tudelft/wdm/group1/stock/event/StockConsumerMultiThreadTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        stockConsumer = new Consumer(stockItemRepository, stockProducer);
+
+        stockItemMilk = new StockItem(5, "Milk", 1);
+        stockItemJuice = new StockItem(4, "Juice", 2);
+        stockItemCoke = new StockItem(3, "Coke", 3);
+    }
+
+
+    @Test
+    public void testHandleOrderCheckedOutWithSufficientStockForMultipleUsers()
```
Comment (qurben) [inline service/stock/src/test/java/nl/tudelft/wdm/group1/stock/event/StockConsumerMultiThreadTest.java]: What does this test exactly test? And how does the concurrency work? When I remove the locks from the consumer the test still passes.

File: service/stock/src/main/java/nl/tudelft/wdm/group1/stock/events/Consumer.java [hunk 2]
```diff
@@ -34,43 +37,57 @@ public void consumeOrderCheckedOut(final Order order)
...
+                }
 
-            try {
-                stockItem.subtractStock(1);
-            } catch (InsufficientStockException | InvalidStockChangeException e) {
-                // we assume the subtraction should not cause any exception
-                throw e;
-            }
+                try {
+                    stockItem.subtractStock(1);
```
Comment (qurben) [inline service/stock/src/main/java/nl/tudelft/wdm/group1/stock/events/Consumer.java]: Subtractstock does not write to the store. This is fixed in #56

**Comment by qurben** [comment]
Because the lock is not shared between threads the lock does not function. Every thread should be able to access the lock for the lock to work.

**Comment by qurben** [comment]
I am unsure about the different read and write locks. When changing from a read lock to a write lock someone else might get the lock during that time. (which happens in my tests.) This can give a situation where multiple threads have seen that an action is possible and then multiple threads actually performing that action.

Having a single lock which first checks if it is possible to do an action and then actually performing the action with the same lock should give no problems.

**Comment by qurben** [comment]
I am unsure why the StockApplicationTest started failing.

**Comment by RUKAIYIN** [comment]
Thanks @qurben, I think everything is fixed and working now

**Pull request merged**